stream-video-android-core/src/main/kotlin/io/getstream/video/android/core/sounds/Sounds.kt

```diff
@@ -16,16 +16,52 @@
 
 package io.getstream.video.android.core.sounds
 
+import android.content.Context
+import android.media.RingtoneManager
+import android.net.Uri
 import androidx.annotation.RawRes
+import io.getstream.log.taggedLogger
 import io.getstream.video.android.core.R
 
 /**
  * Contains all the sounds that the SDK uses.
  *
- * @param incomingCallSound Resource used as a ringtone for incoming calls.
- * @param outgoingCallSound Resource used as a ringing tone for outgoing calls.
+ * @param context Context to be used for retrieving the sounds.
+ * @param incomingCallSoundResId Resource to be used as a ringtone for incoming calls. Set to [DEVICE_INCOMING_RINGTONE] to use the device ringtone.
+ * @param outgoingCallSoundResId Resource to be used as a ringing tone for outgoing calls.
  */
-data class Sounds(
-    @RawRes val incomingCallSound: Int? = R.raw.call_incoming_sound,
-    @RawRes val outgoingCallSound: Int? = R.raw.call_outgoing_sound,
-)
+class Sounds(
+    private val context: Context,
+    @RawRes private val incomingCallSoundResId: Int? = DEVICE_INCOMING_RINGTONE,
+    @RawRes private val outgoingCallSoundResId: Int? = R.raw.call_outgoing_sound,
+) {
+    private val logger by taggedLogger("StreamVideo:Sounds")
+
+    internal val incomingCallSoundUri: Uri?
+        get() = if (incomingCallSoundResId == DEVICE_INCOMING_RINGTONE) {
+            RingtoneManager.getActualDefaultRingtoneUri(context, RingtoneManager.TYPE_RINGTONE) ?: run {
```

Comment:
Also this `getActualDefaultRingtoneUri` are we sure we can always read it?

Reply:
If we can't read it (external storage & no permission etc), then it returns `null` and we fall back to the default incoming sound. Also I've wrapped it in `safeCall` and returned `null` if exception occurs.